### PR TITLE
Feat: Profile Setup (#40012)

### PR DESCRIPTION
**Pull request by bhavya-goel**
*URLs*  :
* for login:
localhost:9000/api/user/login   ( *POST* )
* for user:
localhost:9000/api/user/me  ( *GET* )
**Header token** -
 "key":"Authorization",
"value":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1ZDRkMWNmNWFmYWQxMDFhZjNlNmZmMjYiLCJlbWFpbCI6InVzZXIxQGdtYWlsLmNvbSIsIm5hbWUiOiJ1c2VyMSIsInBhc3N3b3JkIjoiJDJiJDEwJEMuQ2lueDQ1QmlKdVhpeXlhM2diRHV3Skhad25DU29uQVAySG5oMGNqRU5iWXVoNW1haVZTIiwicm9sZSI6InRyYWluZWUiLCJfX3YiOjAsImlhdCI6MTU2NTMzNTk1OH0.T6hy02u8kI-1BZSgK9mZmllJNGaqCz1AX1H8imjkvoQ"

### CODE REVIEW

File: src/controllers/trainee/routes.ts [hunk 1]
```diff
@@ -3,10 +3,9 @@ import { authMiddleWare, validationHandler } from '../../libs';
 import { traineeRoutes } from './Controller';
 import { default as validation } from './validation';
 const traineeRouter = express.Router();
-traineeRouter.route('/:id?')
+traineeRouter.route('/')
                 .get(validationHandler(validation.get), authMiddleWare('getUsers', 'read'), traineeRoutes.get)
                 .post(validationHandler(validation.create), traineeRoutes.create)
```
Comment (nxvinay) [inline src/controllers/trainee/routes.ts]: Why are we not using auth middleware here?

File: src/controllers/trainee/routes.ts [hunk 2]
```diff
@@ -3,10 +3,9 @@ import { authMiddleWare, validationHandler } from '../../libs';
...
-                .put(validationHandler(validation.update), traineeRoutes.update)
-                .delete(validationHandler(validation.delete), traineeRoutes.delete);
-// traineeRouter.delete('/:id?', validationHandler(validation.delete), traineeRoutes.delete);
+                .put(validationHandler(validation.update), traineeRoutes.update);
```
Comment (nxvinay) [inline src/controllers/trainee/routes.ts]: auth middleware?

File: src/controllers/trainee/routes.ts [hunk 3]
```diff
@@ -3,10 +3,9 @@ import { authMiddleWare, validationHandler } from '../../libs';
...
+traineeRouter.delete('/:id?', validationHandler(validation.delete), traineeRoutes.delete);
```
Comment (nxvinay) [inline src/controllers/trainee/routes.ts]: auth middleware?

File: src/controllers/trainee/routes.ts (hunk 1 above)
Comment (bhavya-goel) [reply src/controllers/trainee/routes.ts]: I have added auth middleware to all API requests

**Review by nikitaagg93 — APPROVED** [review]

**Pull request merged**